app/models/course/condition.rb

```diff
@@ -47,15 +47,14 @@ def condition_classes_of(dependent_object)
     #   'Course::Assessment' => ['Course::Condition::Assessment']
     # }
     def dependent_class_to_condition_class_mapping
-      mappings = Course::Condition::ALL_CONDITIONS.map do |condition_name|
-        condition_name.constantize.dependent_classes.map do |dependent_name|
-          { dependent_name => [condition_name] }
-        end
-      end.flatten
-
-      mappings.reduce({}) do |mapping1, mapping2|
-        mapping1.merge(mapping2) { |_, conditions1, conditions2| conditions1 + conditions2 }
+      mappings = Hash.new { |h, k| h[k] = [] }
+
+      Course::Condition::ALL_CONDITIONS.map do |condition_name|
```

Comment:
Currently a little bit ugly but it is cause it is possible for multiple pairs to have the same key. 

E.g. `Course::Achievement => ['Course::Condition::Achievment', 'Course::Condition::SpecialAchievment']`